tests/test_trainer.py

```diff
@@ -999,6 +999,53 @@ def test_trainer_eval_lm(self):
         )
         self.assertEqual(len(dataset), 31)
 
+    @slow
+    @require_datasets
+    def test_tokenizer_without_legacy_files_saving(self):
+        import datasets
+
+        MODEL_ID = "albert-base-v1"
```

Comment:
Initially, I would have liked to create a test for all types of models. However, the development (and testing) would have been longer because it would have meant being sure to have a suitable dataset and task type for each model.

If you ever think it's worth it, I can change it.